envoy/network/connection.h

```diff
@@ -194,8 +194,8 @@ class Connection : public Event::DeferredDeletable,
   /**
    * @return the address provider backing this connection.
    */
-  virtual const SocketAddressProvider& addressProvider() const PURE;
-  virtual SocketAddressProviderSharedPtr addressProviderSharedPtr() const PURE;
+  virtual const ConnectionInfoProvider& addressProvider() const PURE;
```

Comment:
Your plan is to rename the function name and other things in follow ups, right?

/wait-any

Reply:
yes, if we do that in one pr, it will be thousand lines.

Reply:
OK sounds good, thanks.